src/analytics.ts

```diff
@@ -108,30 +108,29 @@ export type TrackingEvent = Readonly<{
     screenName?: string;
     /** Other properties are allowed */
     [key: string]: any;
-}>;
+};
+
+type FirebaseEvent = {
+    name: string;
+    component_type?: string;
+    component_copy?: string;
+    [key: string]: any;
```

Comment:
the only mandatory param is `name`, I've also included `component_type` and `component_copy` because they'll be quite common. In this document there is a list of other params that DA team will use: 
https://confluence.tid.es/pages/resumedraft.action?draftId=125972856&draftShareId=3a1554ba-329b-4a3e-a7d9-169b566a400a&

Reply:
perhaps, instead of `any` this could be more specific: `string | number`, not sure if it makes sense to allow more types

Reply:
I think in ecommerce there are nested structures